compiler/runtime/runtime.h

```diff
@@ -146,25 +146,60 @@ sk_value3_t sk_stack3_pop(sk_stack3_t* st);
 /* The type information exposed by the Skip compiler for each object. */
 /*****************************************************************************/
 
+#define kSkipGcKindClass 0
+#define kSkipGcKindArray 1
+
 // vtable entries created by createGCType in vtable.sk
 typedef struct {
   uint8_t m_refsHintMask;
-  uint8_t m_kind;
-  uint8_t m_tilesPerMask;
+  uint8_t m_kind;  // either kSkipGcKindClass or kSkipGcKindArray
+  uint8_t m_unused_tilesPerMask;
   uint8_t m_hasName;
   uint16_t m_uninternedMetadataByteSize;
-  uint16_t m_internedMetadataByteSize;
-  size_t m_userByteSize;
-#ifdef SKIP32
-  void* unused1;
-  void* unused2;
-#endif
-  void* unused;
-  size_t m_refMask[0];
+  uint16_t m_unused_internedMetadataByteSize;
+  SkipInt m_userByteSize;
+  SkipInt m_unused_padding;
+  SkipInt m_refMask[0];
+  // a 0-terminated name follows if m_hasName is true
 } SKIP_gc_type_t;
 
+/* The uninternedMetadataByteSize is the size preceding the pointer to a
+   non-string GC value. It is:
+   - 1 word for kSkipGcKindClass, its vtable pointer
+   - 2 words for kSkipGcKindArray, its vtable pointer preceded by its size on
+       32 bits, itself preceded by an unused padding of 32 bits on 64-bits arch
+       (see below).
+*/
+#define uninternedMetadataWordSize(ty) ((ty)->m_kind + 1)
+#define uninternedMetadataByteSize(ty) \
+  (uninternedMetadataWordSize(ty) * sizeof(void*))
+
+#define tyIsArray(ty) ((ty)->m_kind)
```

Comment:
This would be clearer and more robust to future changes if it was expressed as a comparison to `kSkipGcKindArray` instead of relying on the macro definitions interpretation as booleans.

Reply:
:+1: